src/XPurchase/VirtoCommerce.XPurchase/CartAggregateRepository.cs

```diff
@@ -68,10 +69,11 @@ public async Task<CartAggregate> GetCartForShoppingCartAsync(ShoppingCart cart,
 
         public async Task<CartAggregate> GetCartAsync(string cartName, string storeId, string userId, string language, string currencyCode, string type = null, string responseGroup = null)
         {
-            var criteria = new CartModule.Core.Model.Search.ShoppingCartSearchCriteria
+            var criteria = new ShoppingCartSearchCriteria
             {
                 StoreId = storeId,
-                CustomerId = userId,
+                // IMPORTANT! Need to specify customerId, otherwise any user cart could be returned while we expect anonymous in this case.
+                CustomerId = userId ?? AnonymousUserId,
```

Comment:
Could you move this existing type https://github.com/VirtoCommerce/vc-module-experience-api/blob/dev/src/XProfile/VirtoCommerce.XProfile/AnonymousUser.cs into core module and use it constant UserName here?

Reply:
Oh no, Anonymous user has a dependency to Security

Reply:
Better define the constant in core  project then